dbms/src/Storages/Page/V3/PageDirectory.cpp

```diff
@@ -104,32 +104,25 @@ void VersionedPageEntries<Trait>::createNewEntry(const PageVersion & ver, const
             assert(last_iter->second.isEntry());
             // It is ok to replace the entry with same sequence and newer epoch, but not valid
             // to replace the entry with newer sequence.
-            if (unlikely(
-                    last_iter->second.being_ref_count.getLatestRefCount() != 1
-                    && last_iter->first.sequence < ver.sequence))
-            {
-                throw Exception(
-                    fmt::format(
-                        "Try to replace normal entry with an newer seq [ver={}] [prev_ver={}] [last_entry={}]",
-                        ver,
-                        last_iter->first,
-                        last_iter->second),
-                    ErrorCodes::LOGICAL_ERROR);
-            }
+            RUNTIME_CHECK_MSG(
+                last_iter->second.being_ref_count.getLatestRefCount() != 1 && last_iter->first.sequence < ver.sequence,
+                "Try to replace normal entry with an newer seq [ver={}] [prev_ver={}] [last_entry={}]",
+                ver,
+                last_iter->first,
+                last_iter->second);
```

Comment:
`RUNTIME_CHECK_MSG` means when the condition is `false`, then throw exception. So you need to rewrite the condition for `if ( something true) throw Exception(...)`

Reply:
fixed